src/thermo/IdealSolidSolnPhaseTabulatedThermo.cpp

```diff
@@ -0,0 +1,332 @@
+/**
+ *  @file IdealSolidSolnPhaseTabulatedThermo.cpp
+ * Definition file for a derived class of ThermoPhase that assumes either
+ * an ideal gas or ideal solution approximation and handles
+ * variable pressure standard state methods for calculating
+ * thermodynamic properties (see \ref thermoprops and
+ * class \link Cantera::IdealSolidSolnPhaseTabulatedThermo IdealSolidSolnPhaseTabulatedThermo\endlink).
+ */
+/*
+ * Copyright (2005) Sandia Corporation. Under the terms of
+ * Contract DE-AC04-94AL85000 with Sandia Corporation, the
+ * U.S. Government retains certain rights in this software.
+ */
+
+#include "cantera/thermo/IdealSolidSolnPhaseTabulatedThermo.h"
+#include "cantera/thermo/PDSS.h"
+#include "cantera/thermo/ThermoFactory.h"
+#include "cantera/base/stringUtils.h"
+#include "cantera/base/ctml.h"
+#include "cantera/thermo/SpeciesThermoFactory.h"
+#include "cantera/thermo/MultiSpeciesThermo.h"
+#include <memory>
+
+namespace Cantera
+{
+IdealSolidSolnPhaseTabulatedThermo::IdealSolidSolnPhaseTabulatedThermo(const IdealSolidSolnPhaseTabulatedThermo& right)
+{
+	*this = right;
+}
+
+IdealSolidSolnPhaseTabulatedThermo& IdealSolidSolnPhaseTabulatedThermo::operator=(const IdealSolidSolnPhaseTabulatedThermo& right)
+{
+	if (&right == this) {
+		return *this;
+	}
+
+	m_h0_RT         = right.m_h0_RT;
+	m_cp0_R         = right.m_cp0_R;
+	m_g0_RT         = right.m_g0_RT;
+	m_s0_R          = right.m_s0_R;
+	m_pp            = right.m_pp;
+	m_xlast			= right.m_xlast;
+
+	return *this;
+
+}
+
+ThermoPhase* IdealSolidSolnPhaseTabulatedThermo::duplMyselfAsThermoPhase() const
+{
+	return new IdealSolidSolnPhaseTabulatedThermo(*this);
+}
+
+void IdealSolidSolnPhaseTabulatedThermo::getPartialMolarEnthalpies(doublereal* result) const {
+	doublereal vdp = (pressure() - m_spthermo.refPressure()) / molarDensity();
+	doublereal rt = temperature() * GasConstant;
+	const vector_fp& h_RT = enthalpy_RT_ref();
+	for (size_t k = 0; k < m_kk; k++) {
+		result[k] = rt*h_RT[k] + vdp;
+	}
+}
+
+void IdealSolidSolnPhaseTabulatedThermo::getPartialMolarEntropies(doublereal* result) const {
+	doublereal xx;
+	doublereal r = GasConstant;
+	const vector_fp& s_R = entropy_R_ref();
+	for (size_t k = 0; k < m_kk; k++) {
+		xx = moleFraction(k);
+		if(xx < SmallNumber) xx = SmallNumber;
+		result[k] = r*(s_R[k] - log(xx));
+	}
+}

```

Comment:
How is this functionally different from the corresponding method in `IdealSolidSolnPhase`?